### PR TITLE
Improvement of two gallery examples regarding categorical colormaps

### LINKED ISSUE #1933: Suggestion for improvement of gallery examples regarding categorical colormaps
In two gallery examples categorical data (without order) is displayed by using color-coding for the categories:
- [Color points by categories](https://www.pygmt.org/dev/gallery/symbols/points_categorical.html#sphx-glr-gallery-symbols-points-categorical-py)
- [3D Scatter plots](https://www.pygmt.org/dev/gallery/3d_plots/scatter3d.html#sphx-glr-gallery-3d-plots-scatter3d-py)

The data (csv file) is loaded into a _pandas_ dataframe and the column of interest is converted to categorical data type. A discrete colormap in categorical format is set up via the `pygmt.makecpt()` function (`+c` through `color_model`). For plotting the symbols with the corresponding fill color the categorical number code is passed through the `color` parameter of the `pygmt.Figure.plot()` method.

I recommend to clarify that by default the labels for the colorbar must be given in **alphabetical** order. If one wants labels in non-alphabetical order, e. g. increasing or decreasing order for displaying categorical data with order, the categorical number code has to be adjusted by the user. For instance, to get '**p**oor, **f**air, **g**ood' instead of '**f**air, **g**ood, **p**oor' (dataframe 'df' with column 'quality'):
``df.quality = df.quality.cat.set_categories(['poor','fair','good'], ordered = True)``
The label order (comma-separated list) passed through `color_model` in the `pygmt.makecpt()` function has no effect on the plotting itself, but affects only the colorbar labels.

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

Fixes #1933 

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by yvonnefroehlich** [comment]
I altered the code as suggested by @seisman in #1933 [comment](https://github.com/GenericMappingTools/pygmt/issues/1933#issuecomment-1139275569).

**Comment by yvonnefroehlich** [comment]
r

**Comment by seisman** [comment]
Ping @michaelgrund and @weiji14 for reviewing this PR, as they wrote the initial version of the example.

File: examples/gallery/3d_plots/scatter3d.py
Comment (seisman) [inline examples/gallery/3d_plots/scatter3d.py:54]: After the revision in 7bcea085a2ffc11bf4a1afe00c74c10d6aa47685, the labels in the `makecpt` call is automatically determined. Do you think we should remove this long paragraph?

File: examples/gallery/3d_plots/scatter3d.py
Comment (seisman) [inline examples/gallery/3d_plots/scatter3d.py:22]: > `df.species = df.species.astype(dtype="category")`

The "species" column is converted to categorical type by this code. I think we can add a comment noting that they are in alphabetical order by default. If anyone wants to adjust the order, they should look at https://pandas.pydata.org/docs/user_guide/categorical.html instead.

File: examples/gallery/3d_plots/scatter3d.py
Comment (yvonnefroehlich) [reply examples/gallery/3d_plots/scatter3d.py:54]: Maybe we can shorten this paragraph a bit and then shift it to the top (see @seisman's second comment).

Comment (seisman) [reply examples/gallery/3d_plots/scatter3d.py:54]: Sounds good.

Comment (yvonnefroehlich) [reply examples/gallery/3d_plots/scatter3d.py]: Done.

**Review by weiji14 — COMMENTED** [review]
Thanks for taking up this Pull Request @yvonnefroehlich!

I've made a few suggested changes to the documentation wording, which you're welcome to "Commit suggestion" directly, or pool several together using "Add suggestion to batch" before committing. See also https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request#applying-a-suggested-change.

image

File: examples/gallery/3d_plots/scatter3d.py
Comment (yvonnefroehlich) [reply examples/gallery/3d_plots/scatter3d.py:55]: Pandas assigns numbers (in an alphabetical manner) to the individual categories, when converting `df.species` to the categorical data type:
```
df.species = df.species.astype(dtype="category")
```
 - [Color points by categories](https://www.pygmt.org/dev/gallery/symbols/points_categorical.html#sphx-glr-gallery-symbols-points-categorical-py): Adelie -> 0, Chinstrap -> 1, Gentoo -> 2
 - [3D Scatter plots](https://www.pygmt.org/dev/gallery/3d_plots/scatter3d.html#sphx-glr-gallery-3d-plots-scatter3d-py): setosa -> 0, versicolor -> 1, virginica -> 2

These numbers are used to define the range of the colormap (`series` of `pygmt.makecpt`):
```
    series=(df.species.cat.codes.min(), df.species.cat.codes.max(), 1),
```
Thus, I think we should use here 'categorical number code'. Also because this term is later used in the comment for the `color` parameter of the `pygmt.Figure.plot()` or `pygmt.Figure.3Dplot()` method:
```
    # Points colored by categorical number code
    color=df.species.cat.codes.astype(int),
```

File: examples/gallery/3d_plots/scatter3d.py
Comment (yvonnefroehlich) [reply examples/gallery/3d_plots/scatter3d.py:26]: For this paragraph this more general formulation should be fine. But maybe we should use ‘categories’ instead of ‘categorical values’, as this is the term used in the [pandas documentation]( https://pandas.pydata.org/docs/user_guide/categorical.html).

File: examples/gallery/3d_plots/scatter3d.py
Comment (seisman) [reply examples/gallery/3d_plots/scatter3d.py:55]: I'm not that familiar with panda's categorical data type, but looking at the documentation and the functions calls `df.species.cat.categories` and `df.species.cat.codes`, I think `categories` and `number code` are better.

File: examples/gallery/symbols/points_categorical.py
Comment (yvonnefroehlich) [reply examples/gallery/symbols/points_categorical.py:24]: There is a typo in the suggested change in line 21:  `you` instead of `ou`.

**Comment by yvonnefroehlich** [comment]
> Thanks for taking up this Pull Request @yvonnefroehlich!

I have to say thank you for the nice and welcoming meeting by @meghanrjones and @weiji14 during the EGU conference! :blush:

Also thanks to @seisman and @weiji14 for their reviews!
But I am a bit unsure how to continue. Should I submit a commit for fixing the typo and interpunction as well as shortening the text (corresponding to the review by @weiji14), but keeping ‘categorical number code’ and ‘categories’ (corresponding to the comment by @seisman)? Or will there be a new review by @weiji14? Or should we wait for the review by @michaelgrund?

File: examples/gallery/3d_plots/scatter3d.py
Comment (weiji14) [reply examples/gallery/3d_plots/scatter3d.py:55]: Hmm you're right, let's go with `categories` and `number code` then. Wasn't aware that pandas actually uses numbers for categorical data!

**Comment by weiji14** [comment]
> Also thanks to @seisman and @weiji14 for their reviews! But I am a bit unsure how to continue. Should I submit a commit for fixing the typo and interpunction as well as shortening the text (corresponding to the review by @weiji14), but keeping ‘categorical number code’ and ‘categories’ (corresponding to the comment by @seisman)? Or will there be a new review by @weiji14? Or should we wait for the review by @michaelgrund?

On second thought, let's keep using 'categories' or `number code' as suggested by @seisman in https://github.com/GenericMappingTools/pygmt/pull/1934#discussion_r890230509 (sorry for the confusion, I was a bit mistaken with pandas categorical data). If @michaelgrund has time, it would be nice to get his input since he wrote this example originally, but if not, we can still continue with the Pull Request.

**Comment by yvonnefroehlich** [comment]
> On second thought, let's keep using 'categories' or `number code' as suggested by @seisman in [#1934 (comment)](https://github.com/GenericMappingTools/pygmt/pull/1934#discussion_r890230509) (sorry for the confusion, I was a bit mistaken with pandas categorical data). If @michaelgrund has time, it would be nice to get his input since he wrote this example originally, but if not, we can still continue with the Pull Request.

No worries and no hurry @weiji14 :wink:
I am totally fine with waiting for the review by @michaelgrund before pushing a further commit.

**Comment by michaelgrund** [comment]
I'm fine with all the changes made so far!

**Comment by yvonnefroehlich** [comment]
> I'm fine with all the changes made so far!

Thanks for your response @michaelgrund !

I updated the text based on the reviews and discussion.

**Review by seisman — APPROVED** [review]
Looks great!

**Review by michaelgrund — APPROVED** [review]

**Review by weiji14 — APPROVED** [review]
Nice one, thanks again @yvonnefroehlich!

**Comment by yvonnefroehlich** [comment]
> Nice one, thanks again @yvonnefroehlich!

I have to say thank you for allowing and helping me to work on this issue / pull request! :blush:




>  We hope that this was a good experience for you. Let us know if there is any way that the contributing process could be improved.

Definitely! :blush: - Likely I learned more than I brought to the project! :wink:

**Pull request merged**
